wpilibcExamples/src/main/cpp/snippets/Encoder/cpp/Robot.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include <numbers>
+
+#include <frc/Encoder.h>
+#include <frc/TimedRobot.h>
+#include <frc/smartdashboard/SmartDashboard.h>
+#include <wpi/deprecated.h>
+
+/**
+ * Encoder snippets for frc-docs.
+ */
+WPI_IGNORE_DEPRECATED
+class Robot : public frc::TimedRobot {
+ public:
+  Robot() {
+    // Configures the encoder to return a distance of 4 for every 256 pulses
+    // Also changes the units of getRate
+    m_encoder.SetDistancePerPulse(4.0 / 256.0);
+    // Configures the encoder to consider itself stopped after .1 seconds
+    m_encoder.SetMaxPeriod(0.1_s);
+    // Configures the encoder to consider itself stopped when its rate is below
+    // 10
+    m_encoder.SetMinRate(10);
+    // Reverses the direction of the encoder
+    m_encoder.SetReverseDirection(true);
+    // Configures an encoder to average its period measurement over 5 samples
+    // Can be between 1 and 127 samples
+    m_encoder.SetSamplesToAverage(5);
+  }
+
+  void TeleopPeriodic() override {
+    // Gets the distance traveled
+    m_encoder.GetDistance();
```

Comment:
Do we want to show return values here?  (I know we'll need to disable unused variable warnings).

Reply:
We don't currently show return values in frc-docs encoder page.

I think the question would be whether you declare the variable inline, which shows the type and would be more useful to new programmers, but isn't what you'd want to do in a periodic function in the snippet, or just set a variable.

I'd tend to keep it exactly as it is in the docs to make it faster to get everything converted to snippets and find/fix breakage, and do improvements later.